src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -712,14 +726,22 @@ bool async_manifest_view::is_empty() const noexcept {
     return _stm_manifest.size() == 0;
 }
 
-bool async_manifest_view::in_archive(async_view_search_query_t o) {
+bool async_manifest_view::in_archive(
+  async_view_search_query_t o, std::optional<cursor_range> range) {
```

Comment:
This also doesn't look very clean from the perspective of readability. The method is called `is_archive` but it's can return `false` for the query which is in the archive if the `range` parameter is right.